### PR TITLE
Add Kubernetes extension (JVM only)

### LINKED ISSUE #303: Create a Camel Kubernetes extension
https://github.com/quarkusio/quarkus/tree/master/extensions/kubernetes-client
https://github.com/apache/camel/tree/master/components/camel-kubernetes

### PR DESCRIPTION
**Pull request by jamesnetherton**
Fixes #303

### CODE REVIEW

File: extensions-jvm/kubernetes/integration-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>native</id>
```
Comment (ppalaga) [inline extensions-jvm/kubernetes/integration-test/pom.xml]: Is this supposed to work in native mode?

Comment (jamesnetherton) [reply extensions-jvm/kubernetes/integration-test/pom.xml]: Good spot - I'll remove that.

I started out intending to have native support but hit some issues, hence decided to fall back on JVM only.

Comment (jamesnetherton) [reply extensions-jvm/kubernetes/integration-test/pom.xml]: If CI passes and you have no further comments, I'd prefer to do a follow up PR that.

Comment (ppalaga) [reply extensions-jvm/kubernetes/integration-test/pom.xml]: Yes, no problem with a followup.

**Pull request merged**